### PR TITLE
Add disassemble benchmarks and more permutations to assemble

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit expands our coverage of the assemble benchmarks by adding
assembly cases with >1 circuit. At the same time this adds a second
benchmark class which adds coverage for qobj disassemble.

### Details and comments

### CODE REVIEW

File: test/benchmarks/assembler.py [hunk 1]
```diff
@@ -16,20 +16,42 @@
...
         seed = 42
         self.circuit = random_circuit(n_qubits, depth, measure=True,
                                       conditional=True, seed=seed)
+        self.circuits = [self.circuit] * number_of_circuits
 
-    def time_assemble_circuit(self, _, __):
-        assemble(self.circuit)
+
+    def time_assemble_circuit(self, _, __, ___):
+        assemble(self.circuits)
```
Comment (kdk) [inline test/benchmarks/assembler.py:39]: Do we have precedent on whether or not we want to set explicit version numbers or let asv manage it?

File: test/benchmarks/assembler.py [hunk 2]
```diff
@@ -16,20 +16,42 @@
...
-    def setup(self, n_qubits, depth):
+    def setup(self, n_qubits, depth, number_of_circuits):
         seed = 42
         self.circuit = random_circuit(n_qubits, depth, measure=True,
                                       conditional=True, seed=seed)
+        self.circuits = [self.circuit] * number_of_circuits
 
-    def time_assemble_circuit(self, _, __):
-        assemble(self.circuit)
+
```
Comment (kdk) [inline test/benchmarks/assembler.py]:
```suggestion
```

**Review by kdk — COMMENTED** [review]
Looks good, one lint failure, one question.

File: test/benchmarks/assembler.py (hunk 1 above)
Comment (mtreinish) [reply test/benchmarks/assembler.py:39]: Typically we've been manually versioning things just to be more explicit about it. I'll add a version now

Comment (mtreinish) [reply test/benchmarks/assembler.py:39]: Done in: https://github.com/Qiskit/qiskit/pull/846/commits/fc32f832deb9587349accc85005a2588a812a5e0

**Review by kdk — APPROVED** [review]

**Pull request merged**